warehouse/query-core/src/main/java/datawave/query/iterator/QueryIterator.java

```diff
@@ -1018,10 +1018,12 @@ public Entry<DocumentData,Document> apply(@Nullable Entry<Key,Document> input) {
             // note that we have already reduced the document to those attributes to keep. This will reduce the attributes further
             // base on those fields we are limiting.
             if (gatherTimingDetails()) {
-                documents = Iterators.transform(documents,
-                                new EvaluationTrackingFunction<>(QuerySpan.Stage.LimitFields, trackingSpan, new LimitFields(this.getLimitFieldsMap())));
+                documents = Iterators.transform(
+                                documents,
+                                new EvaluationTrackingFunction<>(QuerySpan.Stage.LimitFields, trackingSpan, new LimitFields(this.getLimitFieldsMap(), this
+                                                .getMatchingFieldSets())));
             } else {
-                documents = Iterators.transform(documents, new LimitFields(this.getLimitFieldsMap()));
+                documents = Iterators.transform(documents, new LimitFields(this.getLimitFieldsMap(), this.getMatchingFieldSets()));
```

Comment:
Should limit fields be encapsulated? Then you could have something like
```
documents = Iterators.transform(documents, getLimitFields());
```

getLimitFields could look like this
```
getLimitFields() {
    return new LimitFields(this.getLimitFieldsMap(), this.getMatchingFieldSets())
}

# or..
getLimitFields() {
if ( limitFields == null){
    limitFields = new LimitFields(this.getLimitFieldsMap(), this.getMatchingFieldSets());
}
return limitFields
```

Reply:
hmm.... I guess I don't have a problem with that but what does that help us with?